src/script/components/Conversation/Conversation.tsx

```diff
@@ -119,12 +135,104 @@ export const Conversation: FC<ConversationProps> = ({
     }
   }, [readMessagesBuffer.length]);
 
-  if (!mainViewModel) {
-    return null;
-  }
+  const uploadImages = useCallback(
+    (images: File[]) => {
+      if (!activeConversation) {
+        return;
+      }
+
+      if (isHittingUploadLimit(images, repositories.asset)) {
+        return;
+      }
 
-  const {content: contentViewModel} = mainViewModel;
-  const {conversationRepository, repositories} = contentViewModel;
+      for (const image of Array.from(images)) {
+        const isImageTooLarge = image.size > CONFIG.MAXIMUM_IMAGE_FILE_SIZE;
+
+        if (isImageTooLarge) {
+          const isGif = image.type === 'image/gif';
+          const maxSize = CONFIG.MAXIMUM_IMAGE_FILE_SIZE / 1024 / 1024;
+
+          showWarningModal(
+            t(isGif ? 'modalGifTooLargeHeadline' : 'modalPictureTooLargeHeadline'),
+            t(isGif ? 'modalGifTooLargeMessage' : 'modalPictureTooLargeMessage', maxSize),
+          );
+
+          return;
+        }
+      }
+
+      repositories.message.uploadImages(activeConversation, images);
```

Comment:
do we need a new line here too? it looks too spacious to me.

Reply:
I find it more readable, we know its separate functionality